### PR TITLE
Include Response bytes when payload didn't deserialize as expected

### PR DESCRIPTION
**Pull request by epompeii**
The current error handling for `progenatior-client` does not allow for reattempts at deserializing.
This is very useful when creating clients that will need a little more flexibility such as forwards compatibility.

This changeset breaks out the `InvalidResponsePayload` within `progenatior-client` into two separate errors:

- `InvalidResponseBytes`: When `reqwest` fails to actually get the bytes for the response.
- `InvalidResponsePayload`: When the payload didn't deserialize as expected, now also includes the bytes.

### CODE REVIEW

File: progenitor-client/src/progenitor_client.rs [hunk 1]
```diff
@@ -240,9 +240,14 @@ pub enum Error<E = ()> {
     /// A documented, expected error response.
     ErrorResponse(ResponseValue<E>),
 
+    /// An expected response when upgrading connection.
+    InvalidUpgrade(reqwest::Error),
+
+    /// An expected response code whose deserialization failed.
+    InvalidResponseBytes(reqwest::Error),
```
Comment (ahl) [inline progenitor-client/src/progenitor_client.rs:247]: How about:
```suggestion
    /// Encountered an error reading the body for an expected response.
    ResponseBodyError(reqwest::Error),
```

File: progenitor-client/src/progenitor_client.rs [hunk 2]
```diff
@@ -314,7 +323,13 @@ where
...
                 rve.fmt_info(f)
             }
-            Error::InvalidResponsePayload(e) => {
+            Error::InvalidUpgrade(e) => {
+                write!(f, "Invalid Response Upgrade: {}", e)
+            }
+            Error::InvalidResponseBytes(e) => {
+                write!(f, "Invalid Response Body Bytes: {}", e)
+            }
+            Error::InvalidResponsePayload(_b, e) => {
```
Comment (ahl) [inline progenitor-client/src/progenitor_client.rs:332]: we don't want to emit the bytes?

File: progenitor-client/src/progenitor_client.rs [hunk 3]
```diff
@@ -240,9 +240,14 @@ pub enum Error<E = ()> {
     /// A documented, expected error response.
     ErrorResponse(ResponseValue<E>),
 
+    /// An expected response when upgrading connection.
+    InvalidUpgrade(reqwest::Error),
```
Comment (ahl) [inline progenitor-client/src/progenitor_client.rs:244]: Let's move this up to be right after `CommunicationError` which is probably what this should have been initially.

```suggestion
    /// An error occurred when upgrading connection.
    UpgradeError(reqwest::Error),
```

File: docs/progenitor-client.md [hunk 4]
```diff
@@ -88,7 +88,8 @@ pub enum Error<E = ()> {
     InvalidRequest(String),
     CommunicationError(reqwest::Error),
     ErrorResponse(ResponseValue<E>),
```
Comment (ahl) [inline docs/progenitor-client.md:90]: let's add the upgrade error here as well

File: progenitor-client/src/progenitor_client.rs [hunk 5]
```diff
@@ -6,7 +6,7 @@
 
 use std::ops::{Deref, DerefMut};
 
-use bytes::Bytes;
+pub use bytes::Bytes;
```
Comment (ahl) [inline progenitor-client/src/progenitor_client.rs:9]: does this need to be pub? `Error::InvalidResponsePayload`'s inner type isn't `pub`

**Review by ahl — CHANGES_REQUESTED** [review]
This is great; thanks for doing this!

**Comment by epompeii** [comment]
Thanks for all the feedback, @ahl !
I have implemented all the suggested changes. Let me know if you see anything else 😃

**Review by ahl — APPROVED** [review]

**Comment by ahl** [comment]
Thanks for this!

**Pull request merged**